pylint in embedded python interpreter

### Current problem

We're facing 2 issues:
1. pylint exit application when parsing is finished. (Perhaps I missed the option to prevent that.)
2. I made a hack in our application to prevent exit but when I try to re-evaluate the file several times, it only display first evaluated warnings and errors.  
code used:
```
import pylint.lint
pylint_opts = ['myfile.py']
pylint.lint.Run(pylint_opts)
#change myfile.py
pylint.lint.Run(pylint_opts)
```
### Desired solution

Ability to evaluate same file several times in same python session.
Perhaps a workaround would be to create some virtual env to start pylint ?

### Additional context

_No response_

ok, do_exit=False is what I needed for the first point. Thanks.
For second, here is my test case:
test.py is:
```
fct()
```
```
C:\Users\Laurent.Caumont>c:\devtools\Python38\python.exe
Python 3.8.2 (tags/v3.8.2:7b3ab59, Feb 25 2020, 23:03:10) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pylint.lint
>>> pylint_opts = ['-E', r'c:\tmp\test.py']
>>> pylint.lint.Run(pylint_opts, do_exit=False)
************* Module test
c:\tmp\test.py:1:0: E0602: Undefined variable 'fct' (undefined-variable)
<pylint.lint.run.Run object at 0x000002224CCC0AF0>
```
now edit test.py:
```
#fct()
```
in same python session:
```
>>> pylint.lint.Run(pylint_opts, do_exit=False)
************* Module test
c:\tmp\test.py:1:0: E0602: Undefined variable 'fct' (undefined-variable)
<pylint.lint.run.Run object at 0x000002225160FC70>
```
It report the same error.
Should use of reporter fix this issue ?
I was not able to test your code because I don't know what is "text" in "text.TextReporter(pylint_output)"

--- a/doc/user_guide/output.rst
+++ b/doc/user_guide/output.rst
@@ -21,35 +21,6 @@ a colorized report to stdout at the same time:
 
   --output-format=json:somefile.json,colorized
 
-Finally, it is possible to invoke pylint programmatically with a
-reporter initialized with a custom stream:
-
-::
-
-    from io import StringIO
-
-    from pylint.lint import Run
-    from pylint.reporters.text import TextReporter
-
-    pylint_output = StringIO()  # Custom open stream
-    reporter = TextReporter(pylint_output)
-    Run(["test_file.py"], reporter=reporter, do_exit=False)
-    print(pylint_output.getvalue())  # Retrieve and print the text report
-
-The reporter can accept any stream object as as parameter. In this example,
-the stream outputs to a file:
-
-::
-
-    from pylint.lint import Run
-    from pylint.reporters.text import TextReporter
-
-    with open("report.out", "w") as f:
-        reporter = TextReporter(f)
-        Run(["test_file.py"], reporter=reporter, do_exit=False)
-
-This would be useful to capture pylint output in an open stream which
-can be passed onto another program.
 
 Custom message formats
 ''''''''''''''''''''''''''''

--- a/doc/user_guide/output.rst
+++ b/doc/user_guide/output.rst
@@ -26,19 +26,27 @@ reporter initialized with a custom stream:
 
 ::
 
-  pylint_output = StringIO() # Custom open stream
-  reporter = text.TextReporter(pylint_output)
-  Run(["test_file.py"], reporter=reporter, do_exit=False)
-  print(pylint_output.getvalue()) # Retrieve and print the text report
+    from io import StringIO
+
+    from pylint.lint import Run
+    from pylint.reporters.text import TextReporter
+
+    pylint_output = StringIO()  # Custom open stream
+    reporter = TextReporter(pylint_output)
+    Run(["test_file.py"], reporter=reporter, do_exit=False)
+    print(pylint_output.getvalue())  # Retrieve and print the text report
 
 The reporter can accept any stream object as as parameter. In this example,
 the stream outputs to a file:
 
 ::
 
-  with open("report.out", "w") as f:
-    reporter = text.TextReporter(f)
-    Run(["test_file.py"], reporter=reporter, do_exit=False)
+    from pylint.lint import Run
+    from pylint.reporters.text import TextReporter
+
+    with open("report.out", "w") as f:
+        reporter = TextReporter(f)
+        Run(["test_file.py"], reporter=reporter, do_exit=False)
 
 This would be useful to capture pylint output in an open stream which
 can be passed onto another program.

--- a/doc/user_guide/run.rst
+++ b/doc/user_guide/run.rst
@@ -61,9 +61,7 @@ called by the command line. You can either patch ``sys.argv`` or supply argument
 
   sys.argv = ["pylint", "your_file"]
   pylint.run_pylint()
-
   # Or:
-
   pylint.run_pylint(argv=["your_file"])
 
 To silently run Pylint on a ``module_name.py`` module,
@@ -72,6 +70,7 @@ and get its standard output and error:
 .. sourcecode:: python
 
   from pylint import epylint as lint
+
   (pylint_stdout, pylint_stderr) = lint.py_run('module_name.py', return_std=True)
 
 It is also possible to include additional Pylint options in the first argument to ``py_run``:
@@ -79,10 +78,42 @@ It is also possible to include additional Pylint options in the first argument t
 .. sourcecode:: python
 
   from pylint import epylint as lint
+
   (pylint_stdout, pylint_stderr) = lint.py_run('module_name.py --disable C0114', return_std=True)
 
 The options ``--msg-template="{path}:{line}: {category} ({msg_id}, {symbol}, {obj}) {msg}"`` and
 ``--reports=n`` are set implicitly inside the ``epylint`` module.
+
+Finally, it is possible to invoke pylint programmatically with a
+reporter initialized with a custom stream:
+
+.. sourcecode:: python
+
+    from io import StringIO
+
+    from pylint.lint import Run
+    from pylint.reporters.text import TextReporter
+
+    pylint_output = StringIO()  # Custom open stream
+    reporter = TextReporter(pylint_output)
+    Run(["test_file.py"], reporter=reporter, do_exit=False)
+    print(pylint_output.getvalue())  # Retrieve and print the text report
+
+The reporter can accept any stream object as as parameter. In this example,
+the stream outputs to a file:
+
+.. sourcecode:: python
+
+    from pylint.lint import Run
+    from pylint.reporters.text import TextReporter
+
+    with open("report.out", "w") as f:
+        reporter = TextReporter(f)
+        Run(["test_file.py"], reporter=reporter, do_exit=False)
+
+This would be useful to capture pylint output in an open stream which
+can be passed onto another program.
+
 
 Command line options
 --------------------

--- a/doc/user_guide/run.rst
+++ b/doc/user_guide/run.rst
@@ -2,8 +2,8 @@
  Running Pylint
 ================
 
-Invoking Pylint
----------------
+From the command line
+---------------------
 
 Pylint is meant to be called from the command line. The usage is ::
 
@@ -38,6 +38,9 @@ a python package. In order to run pylint over all modules and packages within th
 subtree of a directory, the ``--recursive=y`` option must be provided.
 
 For more details on this see the :ref:`faq`.
+
+From another python program
+---------------------------
 
 It is also possible to call Pylint from another Python program,
 thanks to the ``Run()`` function in the ``pylint.lint`` module

--- a/doc/user_guide/run.rst
+++ b/doc/user_guide/run.rst
@@ -186,24 +186,19 @@ initialization hooks (i.e. the ``--init-hook`` option).
 Exit codes
 ----------
 
-Pylint returns bit-encoded exit codes. If applicable, the table below lists the related
-stderr stream message output.
+Pylint returns bit-encoded exit codes.
 
-=========  =========================  ==========================================
-exit code  meaning                    stderr stream message
-=========  =========================  ==========================================
+=========  =========================
+exit code  meaning
+=========  =========================
 0          no error
 1          fatal message issued
 2          error message issued
 4          warning message issued
 8          refactor message issued
 16         convention message issued
-32         usage error                - "internal error while receiving results\
-                                        from child linter" "Error occurred,
-                                        stopping the linter."
-                                      - "<return of linter.help()>"
-                                      - "Jobs number <#> should be greater \
-                                        than 0"
-                                      - "<IOError message when trying to open \
-                                        output file>"
-=========  =========================  ==========================================
+32         usage error
+=========  =========================
+
+For example, an exit code of ``20`` means there was at least one warning message (4)
+and at least one convention message (16) and nothing else.

